===================
Citing InterProScan
===================

Latest  InterPro and InterProScan publications
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

If you find InterPro useful for your research, please cite the following publications:

InterPro
--------

`InterPro in 2022 <https://academic.oup.com/nar/article/51/D1/D418/6814474?login=false>`_
Paysan-Lafosse T, Blum M, Chuguransky S, Grego T, Pinto BL, Salazar GA, Bileschi ML, Bork P, Bridge A, 
Colwell L, Gough J, Haft DH, Letunić I, Marchler-Bauer A, Mi H, Natale DA, Orengo CA, Pandurangan AP, 
Rivoire C, Sigrist CJA, Sillitoe I, Thanki N, Thomas PD, Tosatto SCE, Wu CH, Bateman A. InterPro in 2022. 
Nucleic Acids Res. 2023 Jan 6;51(D1):D418-D427. doi: 10.1093/nar/gkac993. PMID: 36350672; PMCID: PMC9825450.

InterProScan
-------------

`InterProScan 5: genome-scale protein function classification <https://doi.org/10.1093/bioinformatics/btu031>`_
Philip Jones, David Binns, Hsin-Yu Chang, Matthew Fraser, Weizhong Li, Craig McAnulla, Hamish McWilliam, John Maslen, Alex Mitchell, Gift Nuka, Sebastien Pesseat, Antony F. Quinn, Amaia Sangrador-Vegas, Maxim Scheremetjew, Siew-Yit Yong, Rodrigo Lopez, Sarah Hunter
Bioinformatics (2014), PMID: `24451626 <https://europepmc.org/article/MED/24451626>`_

Member database and third party tool publications
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Member database publications
----------------------------

In alphabetical order:

Citing AntiFam
^^^^^^^^^^^^^^

Eberhardt, R. Y., Haft, D. H., Punta, M., Martin, M., O’Donovan, C., & Bateman, A. (2012). AntiFam: A tool to help identify spurious ORFs in protein annotation. Database, bas003

Citing CATH-Gene3D  
^^^^^^^^^^^^^^^^^^

There are multiple citations for CATH-Gene3D, as this member includes CATH Superfamilies and the Gene3D 
database:

Sillitoe I., Bordin N., Dawson N., Waman V.P., Ashford P., Scholes H.M., Pang C.S.M., Woodridge L., Rauer C., Sen N., Abbasian M., Le Cornu S., Lam S.D., Berka K., Varekova I.H., Svobodova R., Lees J., Orengo C.A. CATH: increased structural coverage of functional space. Nucleic Acids Res. 2021 Jan.

Waman V.P., Bordin N., Alcraft R., Vickerstaff R., Rauer C., Chan Q., Sillitoe I., Yamamori H., Orengo C. CATH 2024: CATH-AlphaFlow Doubles the Number of Structures in CATH and Reveals Nearly 200 New Folds. J Mol Biol.

Orengo C.A., Michie A.D., Jones S., Jones D.T., Swindells M.B., Thornton J.M. CATH–a hierarchic classification of protein domain structures. Structure.

Lewis T.E., Sillitoe I., Dawson N., Lam S.D., Clarke T., Orengo C.A., Lees J.G. Gene3D: Extensive prediction of globular domains in proteins.

Citing CDD
^^^^^^^^^^

Wang J., et al. (2023). “The conserved domain database in 2023”, Nucleic Acids Res.51 (D)384-8

Citing Coils
^^^^^^^^^^^^

Kumar P., Petrenas R., Dawson W.M., Schweke H., Levy E.D., Woolfson D.N. (2023). “CC+: A Searchable Database of Validated Coiled Coils in PDB Structures and AlphaFold2 Models”, Protein Science, Volume 32, Article number e4789.

Citing DeepTMHMM
^^^^^^^^^^^^^^^^

Hallgren J., Tsirigos K.D., Pedersen M.D., Almagro Armenteros J.J., Marcatili P., Nielsen H., Krogh A., Winther O. (2022). “DeepTMHMM predicts alpha and beta transmembrane proteins using deep neural networks”, bioRxiv.

Citing FunFam
^^^^^^^^^^^^^

Scheibenreif L, Littmann M, Orengo C, Rost B. “FunFam protein families improve residue level molecular function prediction.” BMC Bioinformatics. 2019.

Citing HAMAP
^^^^^^^^^^^^

Lima T, Auchincloss AH, Coudert E, Keller G, Michoud K, Rivoire C, Bulliard V, de Castro E, Lachaize C, Baratin D, Phan I, Bougueleret L, Bairoch A. “HAMAP: a database of completely sequenced microbial proteome sets and manually curated microbial protein families in UniProtKB/Swiss-Prot.” Nucleic Acids Res. 2009

Citing MobiDB
^^^^^^^^^^^^^

Piovesan D, Del Conte A, Clementel D, Monzon AM, Bevilacqua M, Aspromonte MC, Iserte JA, Orti FE, Marino-Buslje C, Tosatto SCE. “MobiDB: 10 years of intrinsically disordered proteins.” Nucleic Acids Res. 2022

Citing NCBIFam
^^^^^^^^^^^^^^

No pbulication available.

Citing Panther
^^^^^^^^^^^^^^

Mi H, Muruganujan A, Ebert D, Huang X, Thomas PD. “PANTHER version 14: more genomes, a new PANTHER GO-slim and improvements in enrichment analysis tools.” Nucleic Acids Res. 2019

Citing Pfam
^^^^^^^^^^^

Finn RD, Coggill P, Eberhardt RY, Eddy SR, Mistry J, Mitchell AL, Potter SC, Punta M, Qureshi M, Sangrador-Vegas A, Salazar GA, Tate J, Bateman A. “The Pfam protein families database: towards a more sustainable future.” Nucleic Acids Res. 2016

Citing Phobius
^^^^^^^^^^^^^^

Käll L, Krogh A, Sonnhammer EL. “A combined transmembrane topology and signal peptide prediction method.” J Mol Biol. 2004

Citing PirsR & PirsF
^^^^^^^^^^^^^^^^^^^^

No pbulication available.

Citing PRINTS
^^^^^^^^^^^^^

Attwood TK, Croning MD, Flower DR, Lewis AP, Mabey JE, Scordis P, Selley JN, Wright W. “PRINTS-S: the database formerly known as PRINTS.” Nucleic Acids Res. 2000

Citing PROSITE Patterns & Profiles
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Sigrist CJ, Cerutti L, Hulo N, Gattiker A, Falquet L, Pagni M, Bairoch A, Bucher P. “PROSITE: a documented database using patterns and profiles as motif descriptors.” Brief Bioinform. 2002

Citing SignalP
^^^^^^^^^^^^^^

Teufel, F., Almagro Armenteros, J.J., Johansen, A.R. et al. SignalP 6.0 predicts all five types of signal peptides using protein language models. Nat Biotechnol 40, 1023–1025 (2022). https://doi.org/10.1038/s41587-021-01156-3

Citing SLFD
^^^^^^^^^^^

Holliday G.L., Brown S.D., Akiva E., Mischel D., Hicks M.A., Morris J.H., Huang C.C., Meng E.C., Pegg S.C., Ferrin T.E., Babbitt P.C. Biocuration in the structure-function linkage database: the anatomy of a superfamily. Database (Oxford). 2017 Jan 1;2017(1):bax006. doi: 10.1093/database/bax006. Erratum in: Database (Oxford). 2017 Jan 1;2017:null. PMID: 28365730; PMCID: PMC5467563.

Citing SMART
^^^^^^^^^^^^

Letunic I, Doerks T, Bork P. “SMART: recent updates, new developments and status in 2015.” Nucleic Acids Res. 2015

Citing Superfamily
^^^^^^^^^^^^^^^^^^

Wilson D, Pethica R, Zhou Y, Talbot C, Vogel C, Madera M, Chothia C, Gough J. “SUPERFAMILY–sophisticated comparative genomics, data mining, visualization and phylogeny.” Nucleic Acids Res. 2009

Third party tool publications
-----------------------------

Citing BioPython
^^^^^^^^^^^^^^^^

Used in Panther analyses.

Biopython: Cock, P. J., Antao, T., Chang, J. T., Chapman, B. A., Cox, C. J., Dalke, A., … others. (2009). Biopython: freely available Python tools for computational molecular biology and bioinformatics. Bioinformatics, 25 (11), 1422–1423.

Citing BLAST
^^^^^^^^^^^^

Used for CDD analyses.

BLAST: Altschul, S.F., Gish, W., Miller, W., Myers, E.W., Lipman, D.J. (1990) “Basic local alignment search tool.” J. Mol. Biol. 215:403-4102. Altschul, S.F., Madden, T.L., Schaffer, A.A., Zhang, J., Zhang, Z., Miller, W., Lipman, D.J. (1997) “Gapped BLAST and PSI-BLAST: a new generation of protein database search programs.” Nucleic Acids Res. 25:3389-3402.

Citing Easel
^^^^^^^^^^^^

Used to predict ORFs in nucleic acid sequences.

Citing HMMER3
^^^^^^^^^^^^^

Used in AntiFam, CATH-Gene3D, FunFam, HAMAP, NCBIFam, Panther, Pfam, PirsF, PirsR, SFLD, SMART and Superfamily analyses.

HMMER3: Eddy, S.R. (2011). “Accelerated Profile HMM Searches.” PLoS Comput. Biol. 7(10): e1002195.

Citing TreeGrafter
^^^^^^^^^^^^^^^^^^

Used in Panther analyses.

TreeGrafter: Tang, H., Finn, R.D., Thomas, P.D. (2019). “TreeGrafter: phylogenetic tree-based annotation of proteins with Gene Ontology terms and other annotations.” Bioinformatics, 35 (3): 518-520.


Previous InterPro and InterProScan publications
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

`The InterPro protein families and domains database: 20 years on <https://doi.org/10.1093/nar/gkaa977>`_
Matthias Blum, Hsin-Yu Chang, Sara Chuguransky, Tiago Grego, Swaathi Kandasaamy, Alex Mitchell, Gift Nuka, Typhaine Paysan-Lafosse, Matloob Qureshi, Shriya Raj, Lorna Richardson, Gustavo A Salazar, Lowri Williams, Peer Bork, Alan Bridge, Julian Gough, Daniel H Haft, Ivica Letunic, Aron Marchler-Bauer, Huaiyu Mi, Darren A Natale, Marco Necci, Christine A Orengo, Arun P Pandurangan, Catherine Rivoire, Christian J A Sigrist, Ian Sillitoe, Narmada Thanki, Paul D Thomas, Silvio C E Tosatto, Cathy H Wu, Alex Bateman, Robert D Finn
Nucleic Acids Research (2020), gkaa977, PMID: `33156333 <https://europepmc.org/article/MED/33156333>`_


`InterPro in 2019: improving coverage, classification and access to protein sequence annotations <https://doi.org/10.1093/nar/gky1100>`_
Alex L Mitchell, Teresa K Attwood, Patricia C Babbitt, Matthias Blum, Peer Bork, Alan Bridge, Shoshana D Brown, Hsin-Yu Chang, Sara El-Gebali, Matthew I Fraser, Julian Gough, David R Haft, Hongzhan Huang, Ivica Letunic, Rodrigo Lopez, Aurélien Luciani, Fabio Madeira, Aron Marchler-Bauer, Huaiyu Mi, Darren A Natale, Marco Necci, Gift Nuka, Christine Orengo, Arun P Pandurangan, Typhaine Paysan-Lafosse, Sebastien Pesseat, Simon C Potter, Matloob A Qureshi, Neil D Rawlings, Nicole Redaschi, Lorna J Richardson, Catherine Rivoire, Gustavo A Salazar, Amaia Sangrador-Vegas, Christian J A Sigrist, Ian Sillitoe, Granger G Sutton, Narmada Thanki, Paul D Thomas, Silvio C E Tosatto, Siew-Yit Yong, Robert D Finn
Nucleic Acids Research (2019) Database Issue 47:D351–D360, PMID: `30398656 <https://europepmc.org/article/MED/30398656>`_

`InterPro in 2017 — beyond protein family and domain annotations <https://doi.org/10.1093/nar/gkw1107>`_
Robert D. Finn, Teresa K. Attwood, Patricia C. Babbitt, Alex Bateman, Peer Bork, Alan J. Bridge, Hsin-Yu Chang, Zsuzsanna Dosztányi, Sara El-Gebali, Matthew Fraser, Julian Gough, David Haft, Gemma L. Holliday, Hongzhan Huang, Xiaosong Huang, Ivica Letunic, Rodrigo Lopez, Shennan Lu, Aron Marchler-Bauer, Huaiyu Mi, Jaina Mistry, Darren A. Natale, Marco Necci, Gift Nuka, Christine A. Orengo, Youngmi Park, Sebastien Pesseat, Damiano Piovesan, Simon C. Potter, Neil D. Rawlings, Nicole Redaschi, Lorna Richardson, Catherine Rivoire, Amaia Sangrador-Vegas, Christian Sigrist, Ian Sillitoe, Ben Smithers, Silvano Squizzato, Granger Sutton, Narmada Thanki, Paul D Thomas, Silvio C. E. Tosatto, Cathy H. Wu, Ioannis Xenarios, Lai-Su Yeh, Siew-Yit Young, Alex L. Mitchell
Nucleic Acids Research (2017), Database Issue 45:D190–D199, PMID: `27899635 <https://europepmc.org/article/MED/27899635>`_

.. _go_publication:

`GO annotation in InterPro: why stability does not indicate accuracy in a sea of changing annotation <https://doi.org/10.1093/database/baw027>`_
Sangrador-Vegas A, Mitchell AL, Chang HY, Yong SY, Finn RD
Database: the Journal of Biological Databases and Curation (2016), 1–8, PMID: `26994912 <https://europepmc.org/article/MED/26994912>`_

`The InterPro protein families database: the classification resource after 15 years <https://doi.org/10.1093/nar/gku1243>`_
Alex Mitchell, Hsin-Yu Chang, Louise Daugherty, Matthew Fraser, Sarah Hunter, Rodrigo Lopez, Craig McAnulla, Conor McMenamin, Gift Nuka, Sebastien Pesseat, Amaia Sangrador-Vegas, Maxim Scheremetjew, Claudia Rato, Siew-Yit Yong, Alex Bateman, Marco Punta, Teresa K. Attwood, Christian J.A. Sigrist, Nicole Redaschi, Catherine Rivoire, Ioannis Xenarios, Daniel Kahn, Dominique Guyot, Peer Bork, Ivica Letunic, Julian Gough, Matt Oates, Daniel Haft, Hongzhan Huang, Darren A. Natale, Cathy H. Wu, Christine Orengo, Ian Sillitoe, Huaiyu Mi, Paul D. Thomas, Robert D. Finn
Nucleic Acids Research (2015), Database issue 43:D213-21, PMID: `25428371 <https://europepmc.org/article/MED/25428371>`_

`InterPro in 2011: new developments in the family and domain prediction database <https://doi.org/10.1093/nar/gkr948>`_
Sarah Hunter; Philip Jones; Alex Mitchell; Rolf Apweiler; Teresa K. Attwood; Alex Bateman; Thomas Bernard; David Binns; Peer Bork; Sarah Burge; Edouard de Castro; Penny Coggill; Matthew Corbett; Ujjwal Das; Louise Daugherty; Lauranne Duquenne; Robert D. Finn; Matthew Fraser; Julian Gough; Daniel Haft; Nicolas Hulo; Daniel Kahn; Elizabeth Kelly; Ivica Letunic; David Lonsdale; Rodrigo Lopez; Martin Madera; John Maslen; Craig McAnulla; Jennifer McDowall; Conor McMenamin; Huaiyu Mi; Prudence Mutowo-Muellenet; Nicola Mulder; Darren Natale; Christine Orengo; Sebastien Pesseat; Marco Punta; Antony F. Quinn; Catherine Rivoire; Amaia Sangrador-Vegas; Jeremy D. Selengut; Christian J. A. Sigrist; Maxim Scheremetjew; John Tate; Manjulapramila Thimmajanarthanan; Paul D. Thomas; Cathy H. Wu; Corin Yeats; Siew-Yit Yong
Nucleic Acids Research (2012), Database issue 40:D306–D312, PMID: `22096229 <https://europepmc.org/article/MED/22096229>`_

`Manual GO annotation of predictive protein signatures: the InterPro approach to GO curation <https://doi.org/10.1093/database/bar068>`_
Burge, S., Kelly, E., Lonsdale, D., Mutowo-Muellenet, P., McAnulla, C., Mitchell, A., Sangrador-Vegas, A., Yong, S., Mulder, N., Hunter, S.
Database: the Journal of Biological Databases and Curation (2012), PMID: `22301074 <https://europepmc.org/article/MED/22301074>`_

`The InterPro BioMart: federated query and web service access to the InterPro Resource <https://doi.org/10.1093/database/bar033>`_
Jones P., Binns D., McMenamin C., McAnulla C., Hunter S.
Database: the Journal of Biological Databases and Curation (2011), PMID: `21785143 <https://europepmc.org/article/MED/21785143>`_

`InterPro protein classification <https://doi.org/10.1007/978-1-60761-977-2_3>`_
McDowall J, Hunter S.
Methods Mol Biol. (2011) Database issue 694:37-47, PMID: `21082426 <https://europepmc.org/article/MED/21082426>`_

`InterPro: the integrative protein signature database <https://dx.doi.org/10.1093%2Fnar%2Fgkn785>`_
Hunter S, Apweiler R, Attwood TK, Bairoch A, Bateman A, Binns D, Bork P, Das U, Daugherty L, Duquenne L, Finn RD, Gough J, Haft D, Hulo N, Kahn D, Kelly E, Laugraud A, Letunic I, Lonsdale D, Lopez R, Madera M, Maslen J, McAnulla C, McDowall J, Mistry J, Mitchell A, Mulder N, Natale D, Orengo C, Quinn AF, Selengut JD, Sigrist CJ, Thimma M, Thomas PD, Valentin F, Wilson D, Wu CH, Yeats C.
Nucleic Acids Res. (2009), Database issue 37:D211-5, PMID: `18940856 <https://europepmc.org/article/MED/18940856>`_

`The InterPro database and tools for protein domain analysis <https://doi.org/10.1002/0471250953.bi0207s21>`_
Mulder NJ, Apweiler R.
Curr Protoc Bioinformatics (2008), Chapter 2:Unit 2.7, PMID: `18428686 <https://europepmc.org/article/MED/18428686>`_

`InterPro and InterProScan: tools for protein sequence classification and comparison <https://doi.org/10.1007/978-1-59745-515-2_5>`_
Mulder N, Apweiler R.
Methods Mol Biol (2007), Database issue 396:59-70, PMID: `18025686 <https://europepmc.org/article/MED/18025686>`_

`InterProScan: protein domains identifier <https://doi.org/10.1093/nar/gki442>`_
Quevillon E., Silventoinen V., Pillai S., Harte N., Mulder N., Apweiler R., Lopez R.
Nucleic Acids Research (2005), Vol. 33, Issue suppl 2, PMID: `15980438 <https://europepmc.org/article/MED/15980438>`_

`New developments in the InterPro database <https://dx.doi.org/10.1093%2Fnar%2Fgkl841>`_
Mulder NJ, Apweiler R, Attwood TK, Bairoch A, Bateman A, Binns D, Bork P, Buillard V, Cerutti L, Copley R, Courcelle E, Das U, Daugherty L, Dibley M, Finn R, Fleischmann W, Gough J, Haft D, Hulo N, Hunter S, Kahn D, Kanapin A, Kejariwal A, Labarga A, Langendijk-Genevaux PS, Lonsdale D, Lopez R, Letunic I, Madera M, Maslen J, McAnulla C, McDowall J, Mistry J, Mitchell A, Nikolskaya AN, Orchard S, Orengo C, Petryszak R, Selengut JD, Sigrist CJ, Thomas PD, Valentin F, Wilson D, Wu CH, Yeats C.
Nucleic Acids Research (2005), Database issue 35:D224-8, PMID: `17202162 <https://europepmc.org/article/MED/17202162>`_

`InterPro, progress and status in 2005 <https://doi.org/10.1093/nar/gki106>`_
Mulder NJ, Apweiler R, Attwood TK, Bairoch A, Bateman A, Binns D, Bradley P, Bork P, Bucher P, Cerutti L, Copley R, Courcelle E, Das U, Durbin R, Fleischmann W, Gough J, Haft D, Harte N, Hulo N, Kahn D, Kanapin A, Krestyaninova M, Lonsdale D, Lopez R, Letunic I, Madera M, Maslen J, McDowall J, Mitchell A, Nikolskaya AN, Orchard S, Pagni M, Ponting CP, Quevillon E, Selengut J, Sigrist CJ, Silventoinen V, Studholme DJ, Vaughan R, Wu CH.
Nucleic Acids Res, Database issue 33:D201-5, PMID: `15608177 <https://europepmc.org/article/MED/15608177>`_

`The InterPro Database, 2003 brings increased coverage and new features <https://dx.doi.org/10.1093%2Fnar%2Fgkg046>`_
Mulder NJ, Apweiler R, Attwood TK, Bairoch A, Barrell D, Bateman A, Binns D, Biswas M, Bradley P, Bork P, Bucher P, Copley RR, Courcelle E, Das U, Durbin R, Falquet L, Fleischmann W, Griffiths-Jones S, Haft D, Harte N, Hulo N, Kahn D, Kanapin A, Krestyaninova M, Lopez R, Letunic I, Lonsdale D, Silventoinen V, Orchard SE, Pagni M, Peyruc D, Ponting CP, Selengut JD, Servant F, Sigrist CJ, Vaughan R, Zdobnov EM.
Nucleic Acids Res (2003), 1;31(1):315-8, PMID: `12520011 <https://europepmc.org/article/MED/12520011>`_

`HMM-based databases in InterPro <https://doi.org/10.1093/bib/3.3.236>`_
Bateman A, Haft DH.
Brief Bioinform (2002), 3(3):236-45, PMID: `12230032 <https://europepmc.org/article/MED/12230032>`_

`InterPro: an integrated documentation resource for protein families, domains and functional sites <https://doi.org/10.1093/bib/3.3.225>`_
Mulder NJ, Apweiler R, Attwood TK, Bairoch A, Bateman A, Binns D, Biswas M, Bradley P, Bork P, Bucher P, Copley R, Courcelle E, Durbin R, Falquet L, Fleischmann W, Gouzy J, Griffith-Jones S, Haft D, Hermjakob H, Hulo N, Kahn D, Kanapin A, Krestyaninova M, Lopez R, Letunic I, Orchard S, Pagni M, Peyruc D, Ponting CP, Servant F, Sigrist CJ; InterPro Consortium.
Brief Bioinform (2002), 3(3):225-35, PMID: `12230031 <https://europepmc.org/article/MED/12230031>`_

`Interactive InterPro-based comparisons of proteins in whole genomes <https://doi.org/10.1093/bioinformatics/18.2.374>`_
Kanapin A, Apweiler R, Biswas M, Fleischmann W, Karavidopoulou Y, Kersey P, Kriventseva EV, Mittard V, Mulder N, Oinn T, Phan I, Servant F, Zdobnov E.
Bioinformatics (2002), 18(2):374-5, PMID: `11847096 <https://europepmc.org/article/MED/11847096>`_

`InterProScan — an integration platform for the signature-recognition methods in InterPro <https://doi.org/10.1093/bioinformatics/17.9.847>`_
Zdobnov EM, Apweiler R.
Bioinformatics (2001), 17(9):847-8, PMID: `11590104 <https://europepmc.org/article/MED/11590104>`_

`InterPro — an integrated documentation resource for protein families, domains and functional sites <https://doi.org/10.1093/bioinformatics/16.12.1145>`_
Apweiler R, Attwood TK, Bairoch A, Bateman A, Birney E, Biswas M, Bucher P, Cerutti L, Corpet F, Croning MD, Durbin R, Falquet L, Fleischmann W, Gouzy J, Hermjakob H, Hulo N, Jonassen I, Kahn D, Kanapin A, Karavidopoulou Y, Lopez R, Marx B, Mulder NJ, Oinn TM, Pagni M, Servant F, Sigrist CJ, Zdobnov EM; InterPro Consortium.
Bioinformatics (2000), 16(12):1145-50, PMID: `11159333 <https://europepmc.org/article/MED/11159333>`_

`The InterPro database, an integrated documentation resource for protein families, domains and functional sites <https://dx.doi.org/10.1093%2Fnar%2F29.1.37>`_
Apweiler R, Attwood TK, Bairoch A, Bateman A, Birney E, Biswas M, Bucher P, Cerutti L, Corpet F, Croning MD, Durbin R, Falquet L, Fleischmann W, Gouzy J, Hermjakob H, Hulo N, Jonassen I, Kahn D, Kanapin A, Karavidopoulou Y, Lopez R, Marx B, Mulder NJ, Oinn TM, Pagni M, Servant F, Sigrist CJ, Zdobnov EM.
Nucleic Acids Res (2001), 1;29(1):37-40, PMID: `11125043 <https://europepmc.org/article/MED/11125043>`_
